subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/FileZipInput.java

```diff
@@ -29,10 +32,29 @@
 
 class FileZipInput implements ZipInput {
 
+    public static ZipInput create(File file) {
+        if (isZipFileSafeToUse()) {
+            return new FileZipInput(file);
+        } else {
+            try {
+                return new StreamZipInput(new FileInputStream(file));
+            } catch (FileNotFoundException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+    }
+
+    /**
+     * {@link ZipFile} is more efficient, but causes memory leaks on older Java versions, so we only use it on more recent ones.
+     */
+    private static boolean isZipFileSafeToUse() {
+        return JavaVersion.current().isJava11Compatible();
```

Comment:
is there a bug report that we can attach here?

Reply:
Unfortunately not, we only know it works in recent Java versions